native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/tasks/BuildNativeImageTask.java

```diff
@@ -203,18 +203,18 @@ public void exec() {
         NativeImageOptions options = getOptions().get();
         GraalVMLogger logger = GraalVMLogger.of(getLogger());
 
-        List<String> args = buildActualCommandLineArgs();
```

Comment:
I'm not 100% sure this can be re-ordered. `buildActualCommandLineArgs()` finalizes options, not sure what that means.

Reply:
Finalizing is a safety net. It means that the user won't be allowed to change the value of the options after this call is made.